pros/src/async_runtime/executor.rs

```diff
@@ -0,0 +1,121 @@
+use core::{
+    cell::RefCell,
+    future::Future,
+    pin::Pin,
+    task::{Context, Waker},
+};
+
+use alloc::{boxed::Box, collections::VecDeque, rc::Rc, task::Wake};
+
+use crate::task_local;
+
+use super::{reactor::Reactor, JoinHandle};
+
+task_local! {
```

Comment:
I feel like task_local is a bit of a confusing name because we have both freertos tasks and async tasks.

Reply:
Async tasks are only used internally, also the std feature this is mimicking is reference to the equivalent of freertos tasks. However I do feel it is worth discussing whether we should change the name of task to thread just in general in another pr.

Reply:
Maybe this could be called `os_task_local` for now?

Reply:
That would work.